src/Command/FetchAssetsCommand.php

```diff
@@ -53,6 +53,7 @@ public function __construct(
         Config\Parser\Parser $configParser,
         private readonly Handler\HandlerFactory $handlerFactory,
         private readonly Asset\Definition\AssetDefinitionFactory $assetDefinitionFactory,
+        private bool $hasWarnings = false,
```

Comment:
We should not promote this property via the constructor. There's no need to modify the property's value from outside. Better declare the property outside the constructor.

Reply:
Thanks for the advice. I've not promoted the property to the constructor on purpose. I understand now the unwanted consequences. Thanks.